src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/AnonymizerEngine.cs

```diff
@@ -65,17 +68,38 @@ public static AnonymizerEngine CreateWithFileContext(string configFilePath, stri
         public ITypedElement AnonymizeElement(ITypedElement element, AnonymizerSettings settings = null)
         {
             EnsureArg.IsNotNull(element, nameof(element));
+            try
+            {
+                ElementNode resourceNode = ElementNode.FromElement(element);
+                return resourceNode.Anonymize(_rules, _processors);
+            }
+            catch (AnonymizerConfigurationErrorsException)
+            {
+                throw;
+            }
+            catch
+            {
+                if(_configurationManager.Configuration.processingErrors == ProcessingErrorsOption.Skip)
+                {
+                    return null;
+                }
 
-            ElementNode resourceNode = ElementNode.FromElement(element);
-            return resourceNode.Anonymize(_rules, _processors);
+                throw;
+            }
         }
 
         public Resource AnonymizeResource(Resource resource, AnonymizerSettings settings = null)
         {
             EnsureArg.IsNotNull(resource, nameof(resource));
 
             ValidateInput(settings, resource);
-            var anonymizedResource = AnonymizeElement(resource.ToTypedElement()).ToPoco<Resource>();
+            var anonymizedResource = AnonymizeElement(resource.ToTypedElement())?.ToPoco<Resource>();
+
+            if(anonymizedResource == null)
+            {
+                return null;
```

Comment:
I believe we should let customer aware there's a failure and the error details.

Reply:
After discussion, if customer decide to ignore the exception, the error is no longer exception that we do not need to return error details. And we just return null to indicate skipping it.